src/main/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkVersion.java

```diff
@@ -0,0 +1,118 @@
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents the version of the Cloud SDK.
+ */
+public class CloudSdkVersion implements Comparable<CloudSdkVersion> {
+
+  private final List<Integer> versionComponents;
+  private final String version;
+
+  /**
+   * Constructs a CloudSdkVersion from a version string.
+   * @param version a non-null, nonempty string of the form 130.0.0
+   * @throws NumberFormatException if the string cannot be parsed
+   */
+  public CloudSdkVersion(String version) throws NumberFormatException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(version));
+
+    this.version = version;
+    this.versionComponents = buildVersionComponents(version);
+  }
+
+  @Override
+  public String toString() {
+    return version;
+  }
+
+  @Override
+  public int compareTo(CloudSdkVersion otherVersion) {
+    List<Integer> mine = new ArrayList<>(this.versionComponents);
+    List<Integer> other = new ArrayList<>(otherVersion.getVersionComponents());
+
+    // if both versions don't have the same number of components, pad the smaller one with zeros
+    rightPadZerosUntilSameLength(mine, other);
+
+    // compare version integers from left to right
+    for (int i = 0; i < mine.size(); i++) {
+      int result = mine.get(i).compareTo(other.get(i));
+      if (result != 0) {
+        return result;
+      }
+    }
+    return 0;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(versionComponents);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) {
+      return false;
+    }
+    if (obj == this) {
+      return true;
+    }
+    if (this.getClass() != obj.getClass()) {
+      return false;
+    }
+    CloudSdkVersion otherVersion = (CloudSdkVersion) obj;
+
+    // if both versions don't have the same number of components, pad the smaller one with zeros
+    List<Integer> mine = new ArrayList<>(this.versionComponents);
+    List<Integer> other = new ArrayList<>(otherVersion.getVersionComponents());
+    rightPadZerosUntilSameLength(mine, other);
+
+    return Iterables.elementsEqual(mine, other);
+  }
+
+  protected List<Integer> getVersionComponents() {
+    return versionComponents;
+  }
+
+  private void rightPadZerosUntilSameLength(List<Integer> first, List<Integer> second) {
+    while (first.size() < second.size()) {
+      first.add(0);
+    }
+    while (first.size() > second.size()) {
+      second.add(0);
+    }
+  }
+
+  private List<Integer> buildVersionComponents(String version) throws NumberFormatException {
+    String[] components = version.split("\\.");
+    ImmutableList.Builder builder = ImmutableList.builder();
+    for (String num : components) {
+      builder.add(Integer.parseInt(num));
```

Comment:
do you have a commitment from the cloud SDK team that their component versions will always be \d+(\.d+)*?